bearium/sources/clearinghouse.move

```diff
@@ -16,9 +13,11 @@ module bearium::room {
     use std::vector;
 
     struct Agency has key {
-        registry: Table<address, bool>, // key must be a module address for future hooks
+        registry: Table<address, Dispatch>, // key must be a module address
     }
 
+    struct Dispatch(|address, u64, &mut FungibleAsset, vector<u8>|) has copy, store;
```

Comment:
You can also use an enum, if you find that you want to extend it in the future to have different inputs and outputs

Reply:
Yeah, enum is powerful for extending indeed

Reply:
Hi @gregnazario, I pushed a new commit that uses an enum and a function value for the new feature.
Could you take a look and let me know if this is the right way to handle it? 65f4958cddb87e05173754e10087e832694bad8d